brainreg_segment/tracks/analysis.py

```diff
@@ -91,27 +95,30 @@ def run_track_analysis(
     )
     if summarise_track:
         summary_csv_file = tracks_directory / (track_name + ".csv")
-        analyse_track_anatomy(atlas, spline, summary_csv_file)
+        analyse_track_anatomy(
+            atlas_layer_data, atlas, spline, summary_csv_file
+        )
 
     return spline
 
 
-def analyse_track_anatomy(atlas, spline, file_path):
+def analyse_track_anatomy(atlas_layer_data, atlas, spline, file_path):
     """
     For a given spline, find the atlas region that each
     "segment" is in, and save to csv.
 
+    :param atlas_layer_data: 3D numpy array of the (possibly registered) image
     :param atlas: brainglobe atlas class
     :param spline: numpy array defining the spline interpolation
     :param file_path: path to save the results to
     :param bool verbose: Whether to print the progress
     """
     spline_regions = []
-    for p in spline.tolist():
+    for coord in spline.tolist():
         try:
-            spline_regions.append(
-                atlas.structures[atlas.structure_from_coords(p)]
-            )
+            coord = [int(c) for c in coord]
+            atlas_value = atlas_layer_data[tuple(coord)]
```

Comment:
```suggestion
            coord = (int(c) for c in coord)
            atlas_value = atlas_layer_data[coord]
```
Saves a few characters

Reply:
Doesn't work though :( You can't index with a generator.